### PR TITLE
[Dashboard] Vote contract proposal drawer

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on refactoring the proposal creation UI components in a decentralized application. It replaces the `Drawer` component with a `Sheet` component for better styling and functionality, and improves the form handling for creating proposals.

### Detailed summary
- Deleted `proposal-form.tsx`.
- Modified `metadata-header.tsx` to adjust the `className` for the `Link`.
- Changed layout in `page.tsx` from `Flex` to `div` for better structure.
- Refactored `ProposalButton` in `proposal-button.tsx`:
  - Replaced `Drawer` with `Sheet`.
  - Updated form handling using `react-hook-form`.
  - Added error handling and form validation for the proposal description.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4899?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4899** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4899?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4898** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4898?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4885** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4885?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by MananTank — COMMENTED** [review]
Can we also fix this overflow issue on mobile on proposals page?

File: apps/dashboard/src/contract-ui/tabs/proposals/components/proposal-button.tsx [hunk 1]
```diff
@@ -15,55 +26,93 @@ interface VoteButtonProps {
...
+      <SheetTrigger asChild>
+        <Button
+          colorScheme="primary"
+          leftIcon={<PlusIcon className="size-5" />}
+        >
+          Create Proposal
+        </Button>
+      </SheetTrigger>
+      <SheetContent className="z-[10000] sm:w-[540px] sm:max-w-[90%] lg:w-[700px]">
+        <SheetHeader>
```
Comment (MananTank) [inline apps/dashboard/src/contract-ui/tabs/proposals/components/proposal-button.tsx:48]: Lets increase the max-width on mobile ( < sm ) , there is not enough space to fit cancel + button text

Comment (kien-ngo) [reply apps/dashboard/src/contract-ui/tabs/proposals/components/proposal-button.tsx:48]: Updated and fixed the UIs in other places as well. thanks ^

**Pull request merged**